include/types.h

```diff
@@ -552,26 +560,32 @@ struct retained_format_field {
     // description of format field to be inserted into header
     std::string description;
 
+    // Value type of retained field (int, float)
     RetainedFieldType type;
 
+    // NUMBER of INFO/FORMAT field,
+    // translated from VCF spec
     RetainedFieldNumber number;
 
-    // We use a bool here to sidestep the trouble of setting
-    // the appropriate type for float, int default values
-    bool default_to_zero;
-
-    // Applicable when number ==BASIC
+    // Applicable when number == BASIC
     int count;
 
+    // Handling of "missing" values by using default
+    // values, or leaving as empty, as instructed
+    DefaultValueFiller default_type;
+
+    // Handling of "combining" the same field
+    // from multiple records
     FieldCombinationMethod combi_method;
 
+    // Constructor
     retained_format_field(std::vector<std::string> orig_names_, std::string name_, RetainedFieldType type_,
-        FieldCombinationMethod combi_method_, RetainedFieldNumber number_, int count_=0, bool default_to_zero_=false)
-        : orig_names(orig_names_), name(name_), type(type_), number(number_), default_to_zero(default_to_zero_), count(count_), combi_method(combi_method_) {
+        FieldCombinationMethod combi_method_, RetainedFieldNumber number_, int count_=0, DefaultValueFiller default_type_=DefaultValueFiller::MISSING)
```

Comment:
The default behavior now is to populate with MISSING (which I think is "correct" as a behavior)... But good to get another opinion.